diff --git a/gdb/gdb_thread_db.h b/gdb/gdb_thread_db.h
index e20b415..c770b0c 100644
--- a/gdb/gdb_thread_db.h
+++ b/gdb/gdb_thread_db.h
@@ -203,16 +203,6 @@ typedef struct td_notify
   } u;
 } td_notify_t;
 
-/* Some people still have libc5 or old glibc with no uintptr_t.
-   They lose.  glibc 2.1.3 was released on 2000-02-25, and it has
-   uintptr_t, so it's reasonable to force these people to upgrade.  */
-
-#ifndef HAVE_UINTPTR_T
-#error No uintptr_t available; your C library is too old.
-/* Inhibit further compilation errors after this error.  */
-#define uintptr_t void *
-#endif
-
 /* Structure used to report event.  */
 typedef struct td_event_msg
 {
