diff -urN shorewall-4.4.15-RC1.orig/configfiles/interfaces shorewall-4.4.15-RC1/configfiles/interfaces
--- shorewall-4.4.15-RC1.orig/configfiles/interfaces	2010-11-24 20:23:37.000000000 +0100
+++ shorewall-4.4.15-RC1/configfiles/interfaces	2010-11-28 10:58:25.083550298 +0100
@@ -8,3 +8,5 @@
 #
 ###############################################################################
 #ZONE	INTERFACE	BROADCAST	OPTIONS
+net    eth0            detect          dhcp
+loc    eth1            detect          dhcp
diff -urN shorewall-4.4.15-RC1.orig/configfiles/policy shorewall-4.4.15-RC1/configfiles/policy
--- shorewall-4.4.15-RC1.orig/configfiles/policy	2010-11-24 20:23:37.000000000 +0100
+++ shorewall-4.4.15-RC1/configfiles/policy	2010-11-28 10:59:43.951505215 +0100
@@ -9,3 +9,11 @@
 ###############################################################################
 #SOURCE	DEST	POLICY		LOG	LIMIT:		CONNLIMIT:
 #				LEVEL	BURST		MASK
+loc            net             ACCEPT
+net            all             DROP            info
+# If you want open access to the Internet from your Firewall
+# remove the comment from the following line.
+#fw             net             ACCEPT
+# THE FOLLOWING POLICY MUST BE LAST
+#                                                                            
+all           all             REJECT          info
diff -urN shorewall-4.4.15-RC1.orig/configfiles/zones shorewall-4.4.15-RC1/configfiles/zones
--- shorewall-4.4.15-RC1.orig/configfiles/zones	2010-11-24 20:23:37.000000000 +0100
+++ shorewall-4.4.15-RC1/configfiles/zones	2010-11-28 11:00:28.647060465 +0100
@@ -10,3 +10,6 @@
 #ZONE	TYPE		OPTIONS		IN			OUT
 #					OPTIONS			OPTIONS
 fw	firewall
+net    ipv4
+loc    ipv4
+#dmz   ipv4
diff -urN shorewall-4.4.15-RC1.orig/configfiles/rules shorewall-4.4.15-RC1/configfiles/rules
--- shorewall-4.4.15-RC1.orig/configfiles/rules	2010-11-24 20:23:37.000000000 +0100
+++ shorewall-4.4.15-RC1/configfiles/rules	2010-11-28 11:05:50.199060028 +0100
@@ -12,3 +12,28 @@
 #SECTION ESTABLISHED
 #SECTION RELATED
 SECTION NEW
+#      Accept DNS connections from the firewall to the network
+#      and from the local network to the firewall (in case dnsmasq is running)
+DNS(ACCEPT)   fw          net
+DNS(ACCEPT)   loc         fw
+
+#      Accept SSH connections from the local network for administration
+#
+SSH(ACCEPT)   loc         fw
+
+#      Allow Ping to Firewall
+#
+Ping(ACCEPT)  net         fw
+Ping(ACCEPT)  loc         fw
+#
+#      Allow all ICMP types (including ping) from firewall
+ACCEPT    fw           loc                     icmp
+ACCEPT    fw           net                     icmp
+#      Allow local network to access weblet/webconf
+#
+HTTP(ACCEPT)   loc        fw
+HTTPS(ACCEPT)   loc        fw
+# timeserver (allow syncing with time servers (default: pool.ntp.org))
+NTP(ACCEPT)         fw       net
+# timeserver (allow LAN clients to sync with the time service on the router)
+# NTP(ACCEPT)         loc    fw
diff -urN shorewall-4.4.15-RC1.orig/configfiles/masq shorewall-4.4.15-RC1/configfiles/masq
--- shorewall-4.4.15-RC1.orig/configfiles/masq	2010-11-24 20:23:37.000000000 +0100
+++ shorewall-4.4.15-RC1/configfiles/masq	2010-11-28 11:06:16.651059755 +0100
@@ -9,3 +9,4 @@
 ###############################################################################
 #INTERFACE:DEST		SOURCE		ADDRESS		PROTO	PORT(S)	IPSEC	MARK	USER/
 #											GROUP
+eth0                  192.168.1.0/24
