diff -ur gcc-4.8.2/gcc/config/i386/linux64.h gcc-4.8.2/gcc/config.new/i386/linux64.h
--- gcc-4.8.2/gcc/config/i386/linux64.h	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/i386/linux64.h	2014-01-17 01:22:01.348381395 +0200
@@ -27,6 +27,6 @@
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
 #define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
diff -ur gcc-4.8.2/gcc/config/i386/t-linux64 gcc-4.8.2/gcc/config.new/i386/t-linux64
--- gcc-4.8.2/gcc/config/i386/t-linux64	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/i386/t-linux64	2014-01-17 01:23:35.121891874 +0200
@@ -33,6 +33,6 @@
 comma=,
 MULTILIB_OPTIONS    = $(subst $(comma),/,$(TM_MULTILIB_CONFIG))
 MULTILIB_DIRNAMES   = $(patsubst m%, %, $(subst /, ,$(MULTILIB_OPTIONS)))
-MULTILIB_OSDIRNAMES = m64=../lib64$(call if_multiarch,:x86_64-linux-gnu)
-MULTILIB_OSDIRNAMES+= m32=$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)$(call if_multiarch,:i386-linux-gnu)
+MULTILIB_OSDIRNAMES = m64=../lib$(call if_multiarch,:x86_64-linux-gnu)
+MULTILIB_OSDIRNAMES+= m32=../lib32$(call if_multiarch,:i386-linux-gnu)
 MULTILIB_OSDIRNAMES+= mx32=../libx32$(call if_multiarch,:x86_64-linux-gnux32)
diff -ur gcc-4.8.2/gcc/config/linux.h gcc-4.8.2/gcc/config.new/linux.h
--- gcc-4.8.2/gcc/config/linux.h	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/linux.h	2014-01-17 01:17:52.000000000 +0200
@@ -74,7 +74,7 @@
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
 #define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/lib32/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKERX32 "/lib/ldx32-uClibc.so.0"
 #define BIONIC_DYNAMIC_LINKER "/system/bin/linker"
diff -ur gcc-4.8.2/gcc/config/mips/linux64.h gcc-4.8.2/gcc/config.new/mips/linux64.h
--- gcc-4.8.2/gcc/config/mips/linux64.h	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/mips/linux64.h	2014-01-17 01:17:52.000000000 +0200
@@ -22,10 +22,10 @@
 #define GNU_USER_LINK_EMULATION64 "elf64%{EB:b}%{EL:l}tsmip"
 #define GNU_USER_LINK_EMULATIONN32 "elf32%{EB:b}%{EL:l}tsmipn32"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/lib64/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKERN32 "/lib64/ld-uClibc.so.0"
 #define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32, \
diff -ur gcc-4.8.2/gcc/config/mips/t-linux64 gcc-4.8.2/gcc/config.new/mips/t-linux64
--- gcc-4.8.2/gcc/config/mips/t-linux64	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/mips/t-linux64	2014-01-17 01:24:56.935848007 +0200
@@ -21,6 +21,6 @@
 MIPS_EL = $(if $(filter %el, $(firstword $(subst -, ,$(target)))),el)
 MIPS_SOFT = $(if $(strip $(filter MASK_SOFT_FLOAT_ABI, $(target_cpu_default)) $(filter soft, $(with_float))),soft)
 MULTILIB_OSDIRNAMES = \
-	../lib32$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabin32$(MIPS_SOFT)) \
-	../lib$(call if_multiarch,:mips$(MIPS_EL)-linux-gnu$(MIPS_SOFT)) \
-	../lib64$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabi64$(MIPS_SOFT))
+	../lib64$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabin32$(MIPS_SOFT)) \
+	../lib32$(call if_multiarch,:mips$(MIPS_EL)-linux-gnu$(MIPS_SOFT)) \
+	../lib$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabi64$(MIPS_SOFT))
diff -ur gcc-4.8.2/gcc/config/rs6000/t-linux64 gcc-4.8.2/gcc/config.new/rs6000/t-linux64
--- gcc-4.8.2/gcc/config/rs6000/t-linux64	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/rs6000/t-linux64	2014-01-17 01:21:18.514975574 +0200
@@ -28,5 +28,5 @@
 MULTILIB_OPTIONS    := m64/m32
 MULTILIB_DIRNAMES   := 64 32
 MULTILIB_EXTRA_OPTS := 
-MULTILIB_OSDIRNAMES := m64=../lib64$(call if_multiarch,:powerpc64-linux-gnu)
+MULTILIB_OSDIRNAMES := m64=../lib$(call if_multiarch,:powerpc64-linux-gnu)
 MULTILIB_OSDIRNAMES += m32=$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)$(call if_multiarch,:powerpc-linux-gnu)

diff -ur gcc-4.8.2/gcc/config/s390/linux.h gcc-4.8.2/gcc/config.new/s390/linux.h
--- gcc-4.8.2/gcc/config/s390/linux.h	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/s390/linux.h	2014-01-17 01:17:52.000000000 +0200
@@ -60,7 +60,7 @@
 #define MULTILIB_DEFAULTS { "m31" }
 #endif
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld.so.1"
 #define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 
 #undef  LINK_SPEC
diff -ur gcc-4.8.2/gcc/config/s390/t-linux64 gcc-4.8.2/gcc/config.new/s390/t-linux64
--- gcc-4.8.2/gcc/config/s390/t-linux64	2012-11-21 18:13:28.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/s390/t-linux64	2014-01-17 01:17:52.000000000 +0200
@@ -7,5 +7,5 @@
 
 MULTILIB_OPTIONS = m64/m31
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64$(call if_multiarch,:s390x-linux-gnu)
-MULTILIB_OSDIRNAMES += $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)$(call if_multiarch,:s390-linux-gnu)
+MULTILIB_OSDIRNAMES = ../lib$(call if_multiarch,:s390x-linux-gnu)
+MULTILIB_OSDIRNAMES += ../lib32$(call if_multiarch,:s390-linux-gnu)
diff -ur gcc-4.8.2/gcc/config/sparc/linux64.h gcc-4.8.2/gcc/config.new/sparc/linux64.h
--- gcc-4.8.2/gcc/config/sparc/linux64.h	2013-01-10 22:38:27.000000000 +0200
+++ gcc-4.8.2/gcc/config.new/sparc/linux64.h	2014-01-17 01:26:13.734991223 +0200
@@ -92,8 +92,8 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -201,7 +201,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
