diff -aur ppp-2.4.5.orig/pppd/plugins/pppol2tp/Makefile.linux ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux
--- ppp-2.4.5.orig/pppd/plugins/pppol2tp/Makefile.linux	2009-11-17 00:26:07.000000000 +0200
+++ ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux	2010-08-07 01:14:57.149239004 +0300
@@ -1,6 +1,6 @@
 #CC	= gcc
 COPTS	= -O2 -g
-CFLAGS	= $(COPTS) -I. -I../.. -I../../../include -fPIC
+CFLAGS	= $(COPTS) -I. -I../.. -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
 
diff -aur ppp-2.4.5.orig/pppd/plugins/rp-pppoe/config.h ppp-2.4.5/pppd/plugins/rp-pppoe/config.h
--- ppp-2.4.5.orig/pppd/plugins/rp-pppoe/config.h	2009-11-17 00:26:07.000000000 +0200
+++ ppp-2.4.5/pppd/plugins/rp-pppoe/config.h	2010-08-07 01:14:57.150238823 +0300
@@ -84,7 +84,7 @@
 #define HAVE_NET_IF_H 1
 
 /* Define if you have the <linux/if.h> header file.  */
-#define HAVE_LINUX_IF_H 1
+/* #undef HAVE_LINUX_IF_H */
 
 /* Define if you have the <net/if_dl.h> header file.  */
 /* #undef HAVE_NET_IF_DL_H */
diff -aur ppp-2.4.5.orig/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.5/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.5.orig/pppd/plugins/rp-pppoe/Makefile.linux	2009-11-17 00:26:07.000000000 +0200
+++ ppp-2.4.5/pppd/plugins/rp-pppoe/Makefile.linux	2010-08-07 01:14:57.151238641 +0300
@@ -26,7 +26,7 @@
 RP_VERSION=3.8p
 
 COPTS=-O2 -g
-CFLAGS=$(COPTS) -I../../../include '-DRP_VERSION="$(RP_VERSION)"'
+CFLAGS=$(COPTS) '-DRP_VERSION="$(RP_VERSION)"'
 all: rp-pppoe.so pppoe-discovery
 
 pppoe-discovery: pppoe-discovery.o debug.o
diff -aur ppp-2.4.5.orig/pppd/plugins/rp-pppoe/plugin.c ppp-2.4.5/pppd/plugins/rp-pppoe/plugin.c
--- ppp-2.4.5.orig/pppd/plugins/rp-pppoe/plugin.c	2009-11-17 00:26:07.000000000 +0200
+++ ppp-2.4.5/pppd/plugins/rp-pppoe/plugin.c	2010-08-07 01:15:36.129138153 +0300
@@ -49,6 +49,10 @@
 #include <net/ethernet.h>
 #include <net/if_arp.h>
 #include <linux/ppp_defs.h>
+#ifndef aligned_u64
+/* should be defined in sys/types.h */
+#define aligned_u64 uint64_t __attribute__((aligned(8)))
+#endif
 #include <linux/if_ppp.h>
 #include <linux/if_pppox.h>
 
