diff -aur ppp-2.4.5.old/pppd/plugins/pppoatm/Makefile.linux ppp-2.4.5/pppd/plugins/pppoatm/Makefile.linux
--- ppp-2.4.5.old/pppd/plugins/pppoatm/Makefile.linux	2009-11-17 00:26:07.000000000 +0200
+++ ppp-2.4.5/pppd/plugins/pppoatm/Makefile.linux	2011-10-31 22:47:37.403708158 +0200
@@ -26,7 +26,7 @@
 else
 CFLAGS += -I.
 PLUGIN_OBJS += text2qos.o text2atm.o misc.o ans.o
-LIBS := -lresolv
+LIBS :=
 endif
 
 #*********
