diff -urN ppp-2.4.4.orig/pppd/plugins/pppoatm/pppoatm.c ppp-2.4.4/pppd/plugins/pppoatm/pppoatm.c
--- ppp-2.4.4.orig/pppd/plugins/pppoatm/pppoatm.c	2006-05-21 14:44:41.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/pppoatm/pppoatm.c	2006-06-30 23:00:25.000000000 +0200
@@ -132,9 +132,9 @@
 {
 	int fd;
 	struct atm_qos qos;
-
+/*
 	system ("/sbin/modprobe -q pppoatm");
-
+*/
 	if (!device_got_set)
 		no_device_given_pppoatm();
 	fd = socket(AF_ATMPVC, SOCK_DGRAM, 0);
