diff -Naur nut-2.4.1.old/drivers/apcsmart.h nut-2.4.1/drivers/apcsmart.h
--- nut-2.4.1.old/drivers/apcsmart.h	2009-02-17 11:20:48.000000000 +0200
+++ nut-2.4.1/drivers/apcsmart.h	2009-05-08 12:51:04.000000000 +0300
@@ -23,6 +23,8 @@
 #include "serial.h"
 #include "timehead.h"
 
+#define isprint(c)  (c >= ' ' && c <= '~')
+
 #define APC_TABLE_VERSION	"version 2.1"
 
 /* Basic UPS reply line structure */
diff -Naur nut-2.4.1.old/drivers/serial.c nut-2.4.1/drivers/serial.c
--- nut-2.4.1.old/drivers/serial.c	2009-02-17 11:20:48.000000000 +0200
+++ nut-2.4.1/drivers/serial.c	2009-05-08 12:51:36.000000000 +0300
@@ -30,6 +30,8 @@
 #include <sys/ioctl.h>
 #include <unistd.h>
 
+#define isprint(c)  (c >= ' ' && c <= '~')
+
 #ifdef HAVE_UU_LOCK
 #include <libutil.h>
 #endif
