--- ltmain.sh.orig	2011-06-19 01:40:19.815634575 +0300
+++ ltmain.sh	2011-06-19 01:41:20.045634561 +0300
@@ -2962,8 +2962,8 @@
 	  # At present, this check doesn't affect windows .dll's that
 	  # are installed into $libdir/../bin (currently, that works fine)
 	  # but it's something to keep an eye on.
-	  test "$inst_prefix_dir" = "$destdir" && \
-	    func_fatal_error "error: cannot install \`$file' to a directory not ending in $libdir"
+	  #test "$inst_prefix_dir" = "$destdir" && \
+	  #  func_fatal_error "error: cannot install \`$file' to a directory not ending in $libdir"
 
 	  if test -n "$inst_prefix_dir"; then
 	    # Stick the inst_prefix_dir data into the link command.
