diff -urN shorewall-4.4.20.1.orig/configfiles/start shorewall-4.4.20.1/configfiles/start
--- shorewall-4.4.20.1.orig/configfiles/start	2011-06-07 01:08:29.000000000 +0200
+++ shorewall-4.4.20.1/configfiles/start	2011-06-07 17:03:51.679102163 +0200
@@ -10,4 +10,7 @@
 # information.
 #
 ###############################################################################
+for file in /etc/shorewall/start.d/* ; do
+    run_user_exit $file
+done
 return 0
diff -urN shorewall-4.4.20.1.orig/configfiles/stop shorewall-4.4.20.1/configfiles/stop
--- shorewall-4.4.20.1.orig/configfiles/stop	2011-06-07 01:08:29.000000000 +0200
+++ shorewall-4.4.20.1/configfiles/stop	2011-06-07 17:04:21.139595966 +0200
@@ -10,3 +10,6 @@
 # information.
 #
 ###############################################################################
+for file in /etc/shorewall/stop.d/* ; do
+    run_user_exit $file
+done
\ Kein Zeilenumbruch am Dateiende.
diff -urN shorewall-4.4.20.1.orig/configfiles/shorewall.conf shorewall-4.4.20.1/configfiles/shorewall.conf
--- shorewall-4.4.20.1.orig/configfiles/shorewall.conf	2011-06-07 01:08:29.000000000 +0200
+++ shorewall-4.4.20.1/configfiles/shorewall.conf	2011-06-07 17:06:51.679102156 +0200
@@ -9,7 +9,7 @@
 #		       S T A R T U P   E N A B L E D
 ###############################################################################
 
-STARTUP_ENABLED=No
+STARTUP_ENABLED=Yes
 
 ###############################################################################
 #		              V E R B O S I T Y
@@ -31,7 +31,7 @@
 
 LOGBURST=
 
-LOGFILE=/var/log/messages
+LOGFILE=/var/log/shorewall.log
 
 LOGFORMAT="Shorewall:%s:%s:"
 
@@ -63,7 +63,7 @@
 
 IPSET=
 
-MODULESDIR=
+MODULESDIR=/lib/modules
 
 PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin:/usr/local/sbin
 
@@ -73,7 +73,7 @@
 
 SHOREWALL_SHELL=/bin/sh
 
-SUBSYSLOCK=/var/lock/subsys/shorewall
+SUBSYSLOCK=/var/run/shorewall
 
 TC=
 
@@ -124,7 +124,7 @@
 
 DETECT_DNAT_IPADDRS=No
 
-DISABLE_IPV6=No
+DISABLE_IPV6=Yes
 
 DONT_LOAD=
 
