diff -urN shorewall-4.6.0.1.orig/configfiles/rules shorewall-4.6.0.1/configfiles/rules
--- shorewall-4.6.0.1.orig/configfiles/rules	2014-05-16 16:31:44.000000000 +0200
+++ shorewall-4.6.0.1/configfiles/rules	2014-05-17 14:58:37.361084089 +0200
@@ -15,3 +15,29 @@
 ?SECTION INVALID
 ?SECTION UNTRACKED
 ?SECTION NEW
+
+#      Accept DNS connections from the firewall to the network
+#      and from the local network to the firewall (in case dnsmasq is running)
+DNS(ACCEPT)   fw          net
+DNS(ACCEPT)   loc         fw
+
+#      Accept SSH connections from the local network for administration
+#
+SSH(ACCEPT)   loc         fw
+
+#      Allow Ping to Firewall
+#
+Ping(ACCEPT)  net         fw
+Ping(ACCEPT)  loc         fw
+#
+#      Allow all ICMP types (including ping) from firewall
+ACCEPT    fw           loc                     icmp
+ACCEPT    fw           net                     icmp
+#      Allow local network to access weblet/webconf
+#
+HTTP(ACCEPT)   loc        fw
+HTTPS(ACCEPT)   loc        fw
+# timeserver (allow syncing with time servers (default: pool.ntp.org))
+NTP(ACCEPT)         fw       net
++# timeserver (allow LAN clients to sync with the time service on the router)
+# NTP(ACCEPT)         loc    fw
