diff -urN shorewall-4.5.20.orig/configfiles/policy shorewall-4.5.20/configfiles/policy
--- shorewall-4.5.20.orig/configfiles/policy	2013-08-24 18:37:56.000000000 +0200
+++ shorewall-4.5.20/configfiles/policy	2013-12-08 12:46:48.870742337 +0100
@@ -9,3 +9,12 @@
 ###############################################################################
 #SOURCE	DEST	POLICY		LOG	LIMIT:		CONNLIMIT:
 #				LEVEL	BURST		MASK
+loc     net    ACCEPT
+net     all    DROP
+# If you want open access to the Internet from your Firewall
+# remove the comment from the following line.
+#fw             net             ACCEPT
+# THE FOLLOWING POLICY MUST BE LAST
+#
+all           all             REJECT          NFLOG(4)
+
