diff -urN shorewall-4.5.4.1.orig/configfiles/masq shorewall-4.5.4.1/configfiles/masq
--- shorewall-4.5.4.1.orig/configfiles/masq	2012-05-30 23:37:29.000000000 +0200
+++ shorewall-4.5.4.1/configfiles/masq	2012-06-08 15:27:19.001507086 +0200
@@ -9,3 +9,4 @@
 ######################################################################################################
 #INTERFACE:DEST		SOURCE		ADDRESS		PROTO	PORT(S)	IPSEC	MARK	USER/	SWITCH
 #											GROUP
+eth0              192.168.1.0/24
