diff -urN shorewall-4.4.15.3.orig/Perl/prog.header shorewall-4.4.15.3/Perl/prog.header
--- shorewall-4.4.15.3.orig/Perl/prog.header	2010-12-28 17:58:22.000000000 +0100
+++ shorewall-4.4.15.3/Perl/prog.header	2011-02-13 11:18:48.611164184 +0100
@@ -43,7 +43,7 @@
     fi
 
     if [ $LOG_VERBOSITY -gt 1 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
@@ -59,7 +59,7 @@
     fi
 
     if [ $LOG_VERBOSITY -gt 0 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
@@ -75,7 +75,7 @@
     fi
 
     if [ $LOG_VERBOSITY -ge 0 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
diff -urN shorewall-4.4.15.3.orig/Perl/prog.header6 shorewall-4.4.15.3/Perl/prog.header6
--- shorewall-4.4.15.3.orig/Perl/prog.header6	2010-12-28 17:58:22.000000000 +0100
+++ shorewall-4.4.15.3/Perl/prog.header6	2011-02-13 11:19:31.219164234 +0100
@@ -43,7 +43,7 @@
     fi
 
     if [ $LOG_VERBOSITY -gt 1 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
@@ -59,7 +59,7 @@
     fi
 
     if [ $LOG_VERBOSITY -gt 0 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
@@ -75,7 +75,7 @@
     fi
 
     if [ $LOG_VERBOSITY -ge 0 ]; then
-        timestamp="$(date +'%b %_d %T') "
+        timestamp="$(date +'%b %d %T') "
         echo "${timestamp}$@" >> $STARTUP_LOG
     fi
 }
