--- keepalived-1.2.4/configure.in	2012-07-26 17:53:57.000000000 +0300
+++ keepalived-1.2.4/configure.in	2013-10-05 11:59:25.000000000 +0300
@@ -62,7 +62,6 @@
   ])
 
 dnl ----[ Kernel version check ]----
-CPPFLAGS="$CPPFLAGS -I$kernelinc"
 AC_MSG_CHECKING([for kernel version])
 AC_TRY_RUN([
     #include <stdlib.h>
@@ -172,7 +172,6 @@
 fi
 
 dnl ----[ Checks for kernel VMAC support ]----
-CPPFLAGS="$CPPFLAGS -I$kernelinc"
 AC_MSG_CHECKING([for kernel macvlan support])
 AC_TRY_COMPILE([
     #include <bits/sockaddr.h>
