diff -urN hostapd-2.0.orig/hostapd/hostapd.conf hostapd-2.0/hostapd/hostapd.conf
--- hostapd-2.0.orig/hostapd/hostapd.conf	2013-01-12 16:42:53.000000000 +0100
+++ hostapd-2.0/hostapd/hostapd.conf	2013-02-02 00:58:36.031296487 +0100
@@ -185,8 +185,8 @@
 # Accept/deny lists are read from separate files (containing list of
 # MAC addresses, one per line). Use absolute path name to make sure that the
 # files can be read on SIGHUP configuration reloads.
-#accept_mac_file=/etc/hostapd.accept
-#deny_mac_file=/etc/hostapd.deny
+#accept_mac_file=/etc/hostapd/accept
+#deny_mac_file=/etc/hostapd/deny
 
 # IEEE 802.11 specifies two authentication algorithms. hostapd can be
 # configured to allow both of these or only one. Open system authentication
@@ -642,20 +642,20 @@
 # Path for EAP server user database
 # If SQLite support is included, this can be set to "sqlite:/path/to/sqlite.db"
 # to use SQLite database instead of a text file.
-#eap_user_file=/etc/hostapd.eap_user
+#eap_user_file=/etc/hostapd/eap_user
 
 # CA certificate (PEM or DER file) for EAP-TLS/PEAP/TTLS
-#ca_cert=/etc/hostapd.ca.pem
+#ca_cert=/etc/hostapd/ca.pem
 
 # Server certificate (PEM or DER file) for EAP-TLS/PEAP/TTLS
-#server_cert=/etc/hostapd.server.pem
+#server_cert=/etc/hostapd/server.pem
 
 # Private key matching with the server certificate for EAP-TLS/PEAP/TTLS
 # This may point to the same file as server_cert if both certificate and key
 # are included in a single file. PKCS#12 (PFX) file (.p12/.pfx) can also be
 # used by commenting out server_cert and specifying the PFX file as the
 # private_key.
-#private_key=/etc/hostapd.server.prv
+#private_key=/etc/hostapd/server.prv
 
 # Passphrase for private key
 #private_key_passwd=secret passphrase
@@ -680,8 +680,8 @@
 # is in DSA parameters format, it will be automatically converted into DH
 # params. This parameter is required if anonymous EAP-FAST is used.
 # You can generate DH parameters file with OpenSSL, e.g.,
-# "openssl dhparam -out /etc/hostapd.dh.pem 1024"
-#dh_file=/etc/hostapd.dh.pem
+# "openssl dhparam -out /etc/hostapd/dh.pem 1024"
+#dh_file=/etc/hostapd/dh.pem
 
 # Fragment size for EAP methods
 #fragment_size=1400
@@ -828,7 +828,7 @@
 # multiple BSSIDs or SSIDs. Each line in this text file is defining a new
 # interface and the line must include VLAN ID and interface name separated by
 # white space (space or tab).
-#vlan_file=/etc/hostapd.vlan
+#vlan_file=/etc/hostapd/vlan
 
 # Interface where 802.1q tagged packets should appear when a RADIUS server is
 # used to determine which VLAN a station is on.  hostapd creates a bridge for
@@ -901,7 +901,7 @@
 
 # File name of the RADIUS clients configuration for the RADIUS server. If this
 # commented out, RADIUS server is disabled.
-#radius_server_clients=/etc/hostapd.radius_clients
+#radius_server_clients=/etc/hostapd/radius_clients
 
 # The UDP port number for the RADIUS authentication server
 #radius_server_auth_port=1812
@@ -938,7 +938,7 @@
 # of (PSK,MAC address) pairs. This allows more than one PSK to be configured.
 # Use absolute path name to make sure that the files can be read on SIGHUP
 # configuration reloads.
-#wpa_psk_file=/etc/hostapd.wpa_psk
+#wpa_psk_file=/etc/hostapd/wpa_psk
 
 # Optionally, WPA passphrase can be received from RADIUS authentication server
 # This requires macaddr_acl to be set to 2 (RADIUS)
@@ -1236,7 +1236,7 @@
 # with pre-configured attributes. This is similar to extra_cred file format,
 # but the AP Settings attributes are not encapsulated in a Credential
 # attribute.
-#ap_settings=hostapd.ap_settings
+#ap_settings=hostapd/ap_settings
 
 # WPS UPnP interface
 # If set, support for external Registrars is enabled.
