diff -aur linux-2.6.35.8/tools/perf/builtin-sched.c linux-2.6.35.8/tools/perf/builtin-sched.c
--- linux-2.6.35.8/tools/perf/builtin-sched.c	2010-10-29 07:52:43.000000000 +0300
+++ linux-2.6.35.8/tools/perf/builtin-sched.c	2010-12-11 20:15:16.000000000 +0200
@@ -478,6 +478,7 @@
 	BUG_ON(ret);
 
 	goto again;
+	return NULL;
 }
 
 static void create_tasks(void)
diff -aur linux-2.6.35.8/tools/perf/Makefile linux-2.6.35.8/tools/perf/Makefile
--- linux-2.6.35.8/tools/perf/Makefile	2010-10-29 07:52:43.000000000 +0300
+++ linux-2.6.35.8/tools/perf/Makefile	2010-12-11 20:36:34.000000000 +0200
@@ -291,11 +291,6 @@
 	BITBUCKET = .perf.dev.null
 endif
 
-ifeq ($(shell sh -c "echo 'int foo(void) {char X[2]; return 3;}' | $(CC) -x c -c -Werror -fstack-protector-all - -o $(BITBUCKET) "$(QUIET_STDERR)" && echo y"), y)
-  CFLAGS := $(CFLAGS) -fstack-protector-all
-endif
-
-
 ### --- END CONFIGURATION SECTION ---
 
 # Those must not be GNU-specific; they are shared with perl/ which may
@@ -537,10 +532,6 @@
 endif
 
 ifeq ($(shell sh -c "(echo '\#include <libelf.h>'; echo 'int main(void) { Elf * elf = elf_begin(0, ELF_C_READ, 0); return (long)elf; }') | $(CC) -x c - $(ALL_CFLAGS) -o $(BITBUCKET) $(ALL_LDFLAGS) $(EXTLIBS) "$(QUIET_STDERR)" && echo y"), y)
-ifneq ($(shell sh -c "(echo '\#include <gnu/libc-version.h>'; echo 'int main(void) { const char * version = gnu_get_libc_version(); return (long)version; }') | $(CC) -x c - $(ALL_CFLAGS) -o $(BITBUCKET) $(ALL_LDFLAGS) $(EXTLIBS) "$(QUIET_STDERR)" && echo y"), y)
-	msg := $(error No gnu/libc-version.h found, please install glibc-dev[el]/glibc-static);
-endif
-
 	ifneq ($(shell sh -c "(echo '\#include <libelf.h>'; echo 'int main(void) { Elf * elf = elf_begin(0, ELF_C_READ_MMAP, 0); return (long)elf; }') | $(CC) -x c - $(ALL_CFLAGS) -o $(BITBUCKET) $(ALL_LDFLAGS) $(EXTLIBS) "$(QUIET_STDERR)" && echo y"), y)
 		BASIC_CFLAGS += -DLIBELF_NO_MMAP
 	endif
diff -aur linux-2.6.35.8/tools/perf/util/cache.h linux-2.6.35.8/tools/perf/util/cache.h
--- linux-2.6.35.8/tools/perf/util/cache.h	2010-10-29 07:52:43.000000000 +0300
+++ linux-2.6.35.8/tools/perf/util/cache.h	2010-12-11 20:15:16.000000000 +0200
@@ -81,6 +81,4 @@
 extern char *perf_pathdup(const char *fmt, ...)
 	__attribute__((format (printf, 1, 2)));
 
-extern size_t strlcpy(char *dest, const char *src, size_t size);
-
 #endif /* __PERF_CACHE_H */
diff -aur linux-2.6.35.8/tools/perf/util/svghelper.c linux-2.6.35.8/tools/perf/util/svghelper.c
--- linux-2.6.35.8/tools/perf/util/svghelper.c	2010-10-29 07:52:43.000000000 +0300
+++ linux-2.6.35.8/tools/perf/util/svghelper.c	2010-12-11 20:15:16.000000000 +0200
@@ -43,11 +43,11 @@
 	return cpu2slot(cpu) * SLOT_MULT;
 }
 
-static double time2pixels(u64 time)
+static double time2pixels(u64 drawtime)
 {
 	double X;
 
-	X = 1.0 * svg_page_width * (time - first_time) / (last_time - first_time);
+	X = 1.0 * svg_page_width * (drawtime - first_time) / (last_time - first_time);
 	return X;
 }
 
