diff -urN syslinux-4.06.orig/Makefile syslinux-4.06/Makefile
--- syslinux-4.06.orig/Makefile	2012-10-23 11:53:17.000000000 +0200
+++ syslinux-4.06/Makefile	2013-01-07 19:06:27.588235222 +0100
@@ -57,7 +57,7 @@
 	   diag libinstaller dos win32 win64 dosutil
 ITARGET  =
 IOBJECTS = $(ITARGET) \
-	utils/gethostip utils/isohybrid utils/mkdiskimage \
+	utils/gethostip utils/mkdiskimage \
 	mtools/syslinux linux/syslinux extlinux/extlinux
 ISUBDIRS = libinstaller mtools linux extlinux utils
 
