diff -u snarf-7.0.orig/configure.in snarf-7.0.new/configure.in
--- snarf-7.0.orig/configure.in	2000-08-09 03:35:29.000000000 +0300
+++ snarf-7.0.new/configure.in	2011-11-24 00:14:37.000000000 +0200
@@ -1,10 +1,10 @@
 dnl Process this file with autoconf to produce a configure script.
 
 dnl This is elrond's optional library check
-AC_DEFUN(FP_CHECK_OPT_LIB,
+AC_DEFUN([FP_CHECK_OPT_LIB],
          [AC_CHECK_FUNC([$2], :,
-                        AC_CHECK_LIB($@)dnl
-                       )dnl
+                        [AC_CHECK_LIB($@)dnl
+                        ])dnl
          ])dnl
 
 
Common subdirectories: snarf-7.0.orig/CVS and snarf-7.0.new/CVS
Common subdirectories: snarf-7.0.orig/debian and snarf-7.0.new/debian
