*** kismet.conf.orig	2010-09-27 20:20:28.000000000 +0100
--- kismet.conf	2010-09-27 20:36:25.000000000 +0100
***************
*** 7,16 ****
  version=2009-newcore
  
  # Name of server (Purely for organizational purposes)
! servername=Kismet_2009
  
  # Prefix of where we log (as used in the logtemplate later)
  # logprefix=/some/path/to/logs
  
  # Do we process the contents of data frames?  If this is enabled, data
  # frames will be truncated to the headers only immediately after frame type
--- 7,17 ----
  version=2009-newcore
  
  # Name of server (Purely for organizational purposes)
! servername=LEAF
  
  # Prefix of where we log (as used in the logtemplate later)
  # logprefix=/some/path/to/logs
+ logprefix=/var/log/kismet
  
  # Do we process the contents of data frames?  If this is enabled, data
  # frames will be truncated to the headers only immediately after frame type
***************
*** 29,34 ****
--- 30,36 ----
  # ncsource=wlan0
  # ncsource=wifi0:type=madwifi
  # ncsource=wlan0:name=intel,hop=false,channel=11
+ ncsource=wlan0
  
  # Comma-separated list of sources to enable.  This is only needed if you defined
  # multiple sources and only want to enable some of them.  By default, all defined
***************
*** 122,132 ****
  # IEEE OUI file used to look up manufacturer info.  We default to the
  # wireshark one since most people have that.
  ouifile=/etc/manuf
! ouifile=/usr/share/wireshark/wireshark/manuf
! ouifile=/usr/share/wireshark/manuf
  
  # Do we have a GPS?
! gps=true
  # Do we use a locally serial attached GPS, or use a gpsd server?
  # (Pick only one)
  gpstype=gpsd
--- 124,134 ----
  # IEEE OUI file used to look up manufacturer info.  We default to the
  # wireshark one since most people have that.
  ouifile=/etc/manuf
! #ouifile=/usr/share/wireshark/wireshark/manuf
! #ouifile=/usr/share/wireshark/manuf
  
  # Do we have a GPS?
! gps=false
  # Do we use a locally serial attached GPS, or use a gpsd server?
  # (Pick only one)
  gpstype=gpsd
