diff -urN ulogd-1.24.orig/configure ulogd-1.24/configure
--- ulogd-1.24.orig/configure	2006-01-25 12:15:22.000000000 +0100
+++ ulogd-1.24/configure	2010-09-29 16:09:42.061641287 +0200
@@ -1728,7 +1728,7 @@
 EOF
 
         MYSQLINCLUDES=`$d/mysql_config --include`
-        MYSQLLIBS=`$d/mysql_config --libs`
+        MYSQLLIBS=`$d/mysql_config --libs | sed 's/-rdynamic//'` 
 
 	DATABASE_DIR="${DATABASE_DIR} mysql"
 
diff -urN ulogd-1.24.orig/configure.in ulogd-1.24/configure.in
--- ulogd-1.24.orig/configure.in	2005-11-25 20:58:27.000000000 +0100
+++ ulogd-1.24/configure.in	2010-09-29 16:11:47.257659473 +0200
@@ -60,7 +60,7 @@
 else
         AC_DEFINE(HAVE_MYSQL)
         MYSQLINCLUDES=`$d/mysql_config --include`
-        MYSQLLIBS=`$d/mysql_config --libs`
+        MYSQLLIBS=`$d/mysql_config --libs | sed 's/-rdynamic//'`
 
 	DATABASE_DIR="${DATABASE_DIR} mysql"
 
diff -urN ulogd-1.24.orig/Makefile.in ulogd-1.24/Makefile.in
--- ulogd-1.24.orig/Makefile.in	2006-01-25 12:26:51.000000000 +0100
+++ ulogd-1.24/Makefile.in	2010-09-29 16:10:32.885674779 +0200
@@ -63,7 +63,7 @@
 	@for d in $(SUBDIRS); do if ! make -C $$d; then exit 1; fi; done
 
 ulogd: ulogd.c $(LIBIPULOG) include/ulogd/ulogd.h conffile/conffile.o $(LIBIPULOG)/libipulog.a ulogd.conf
-	$(CC) $(CFLAGS) -rdynamic $< conffile/conffile.o $(LIBIPULOG)/libipulog.a -o $@ $(LIBS)
+	$(CC) $(CFLAGS) $< conffile/conffile.o $(LIBIPULOG)/libipulog.a -o $@ $(LIBS)
 
 edit = sed -e 's,@libdir\@,$(ULOGD_LIB_PATH),g'
 
