diff -urN ulogd-1.24.orig/configure.in ulogd-1.24/configure.in
--- ulogd-1.24.orig/configure.in	2005-11-25 20:58:27.000000000 +0100
+++ ulogd-1.24/configure.in	2006-02-15 21:54:48.000000000 +0100
@@ -83,7 +83,7 @@
 
 	AC_MSG_CHECKING(for mysql_real_escape_string support)
 
-	MYSQL_FUNCTION_TEST=`strings ${MYSQLLIBS}/libmysqlclient.so | grep mysql_real_escape_string`
+	MYSQL_FUNCTION_TEST=`strings ${dir}/lib/mysql/libmysqlclient.so | grep mysql_real_escape_string`
 
 	if test "x$MYSQL_FUNCTION_TEST" = x
 	then
