diff -urN ulogd-1.24.orig/mysql/Makefile.in ulogd-1.24/mysql/Makefile.in
--- ulogd-1.24.orig/mysql/Makefile.in	2005-11-25 20:58:27.000000000 +0100
+++ ulogd-1.24/mysql/Makefile.in	2012-08-17 15:02:15.874461828 +0200
@@ -14,7 +14,8 @@
 distrib:
 
 $(SHARED_LIBS): %.so: %_sh.o
-	$(LD) -shared $(MYSQL_LDFLAGS) -o $@ $< -lc
+	#$(LD) -shared $(MYSQL_LDFLAGS) -o $@ $< -lc
+	$(CC) -nostartfiles -shared $(MYSQL_LDFLAGS) -o $@ $<
 
 %_sh.o: %.c
 	$(CC) $(MYSQL_CFLAGS) $(SH_CFLAGS) -o $@ -c $<
