diff -aur gcc-3.3.3.orig/gcc/collect2.c gcc-3.3.3/gcc/collect2.c
--- gcc-3.3.3.orig/gcc/collect2.c	2010-03-21 16:10:37.000000000 +0200
+++ gcc-3.3.3/gcc/collect2.c	2010-03-21 16:41:26.000000000 +0200
@@ -1572,7 +1572,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0644);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
