*** uClibc-0.9.30.3/libpthread/linuxthreads/pthread.c.orig	2010-03-12 19:32:42.000000000 +0000
--- uClibc-0.9.30.3/libpthread/linuxthreads/pthread.c	2010-10-29 10:49:29.000000000 +0100
***************
*** 223,229 ****
        mask with e.g. sigsetjmp before creating the first thread);
     - a regular function called from pthread_create when needed. */
  
! static void pthread_initialize(void) __attribute__((constructor));
  
  #ifndef HAVE_Z_NODELETE
  extern void *__dso_handle __attribute__ ((weak));
--- 223,229 ----
        mask with e.g. sigsetjmp before creating the first thread);
     - a regular function called from pthread_create when needed. */
  
! static void pthread_initialize(void);
  
  #ifndef HAVE_Z_NODELETE
  extern void *__dso_handle __attribute__ ((weak));
