diff -urN drivers/net/wireless/ath.orig/regd.c drivers/net/wireless/ath/regd.c
--- drivers/net/wireless/ath.orig/regd.c	2011-08-01 22:56:51.000000000 +0200
+++ drivers/net/wireless/ath/regd.c	2012-12-06 08:10:23.000000000 +0100
@@ -20,6 +20,15 @@
 #include "regd.h"
 #include "regd_common.h"
 
+/* 
+ * add a module parameter to control regulatory domain 
+ * use US as fallback as before
+ * erich.titl@think.ch 12/2012
+ */
+#include <linux/moduleparam.h>
+static uint ctry_default = CTRY_UNITED_STATES;
+module_param(ctry_default,uint,S_IRWUSR | S_IRGRP | S_IROTH );
+
 /*
  * This is a set of common rules used by our world regulatory domains.
  * We have 12 world regulatory domains. To save space we consolidate
@@ -516,7 +525,7 @@
 	    regdmn == CTRY_DEFAULT) {
 		printk(KERN_DEBUG "ath: EEPROM indicates default "
 		       "country code should be used\n");
-		reg->country_code = CTRY_UNITED_STATES;
+		reg->country_code = ctry_default;
 	}
 
 	if (reg->country_code == CTRY_DEFAULT) {
