*** Bering-3.10.18.config	2013-12-05 14:08:37.555162431 +0100
--- Bering-3.10.18.config-i686	2013-12-05 14:08:56.587163299 +0100
***************
*** 2,8 ****
  # Automatically generated file; DO NOT EDIT.
  # Linux/i386 3.10.18 Kernel Configuration
  #
! # CONFIG_64BIT is not set
  CONFIG_X86_32=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
--- 2,8 ----
  # Automatically generated file; DO NOT EDIT.
  # Linux/i386 3.10.18 Kernel Configuration
  #
! # CONFIG_64BIT is not set 
  CONFIG_X86_32=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
