*** Bering-3.4.37.config	2013-03-24 13:34:37.137816131 +0100
--- Bering-3.4.37.config-i686	2013-03-24 13:39:14.865828775 +0100
***************
*** 3,9 ****
  # Linux/x86_64 3.4.37 Kernel Configuration
  #
  CONFIG_64BIT=y
! # CONFIG_X86_32 is not set
  CONFIG_X86_64=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
--- 3,9 ----
  # Linux/x86_64 3.4.37 Kernel Configuration
  #
  CONFIG_64BIT=y
! # CONFIG_X86_32 is not set 
  CONFIG_X86_64=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
