*** Bering-3.10.25.config	2014-01-02 13:46:59.167431239 +0100
--- Bering-3.10.25.config-i686	2014-01-02 13:47:17.631432080 +0100
***************
*** 2,8 ****
  # Automatically generated file; DO NOT EDIT.
  # Linux/i386 3.10.25 Kernel Configuration
  #
! # CONFIG_64BIT is not set
  CONFIG_X86_32=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
--- 2,8 ----
  # Automatically generated file; DO NOT EDIT.
  # Linux/i386 3.10.25 Kernel Configuration
  #
! # CONFIG_64BIT is not set 
  CONFIG_X86_32=y
  CONFIG_X86=y
  CONFIG_INSTRUCTION_DECODER=y
