From 4cbea24a9e2a71f6586a107c1e5ad4f2dc85b871 Mon Sep 17 00:00:00 2001
From: Kozlov Dmitry <xeb@mail.ru>
Date: Thu, 1 Nov 2012 18:06:28 +0400
Subject: [PATCH] cli: terminate username: check for username is not NULL

---
 accel-pppd/cli/std_cmd.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/accel-pppd/cli/std_cmd.c b/accel-pppd/cli/std_cmd.c
index d5259a5..97ab8d7 100644
--- a/accel-pppd/cli/std_cmd.c
+++ b/accel-pppd/cli/std_cmd.c
@@ -128,6 +128,8 @@ static int terminate_exec1(char * const *f, int f_cnt, void *cli)
 
 	pthread_rwlock_rdlock(&ppp_lock);
 	list_for_each_entry(ppp, &ppp_list, entry) {
+		if (!ppp->username)
+			continue;
 		if (pcre_exec(re, NULL, ppp->username, strlen(ppp->username), 0, 0, NULL, 0) < 0)
 			continue;
 		if (hard)
-- 
1.7.4.1

From 5fd5776d2175e92179461fd69f897146c151c765 Mon Sep 17 00:00:00 2001
From: Kozlov Dmitry <xeb@mail.ru>
Date: Sat, 3 Nov 2012 02:11:07 +0400
Subject: [PATCH] cli: fix 'show sessions match' without enough arguments

---
 accel-pppd/cli/show_sessions.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/accel-pppd/cli/show_sessions.c b/accel-pppd/cli/show_sessions.c
index c0fec58..983e620 100644
--- a/accel-pppd/cli/show_sessions.c
+++ b/accel-pppd/cli/show_sessions.c
@@ -148,7 +148,7 @@ static int show_ses_exec(const char *cmd, char * const *f, int f_cnt, void *cli)
 				return CLI_CMD_OK;
 			}
 		} else if (!strcmp(f[i], "match")) {
-			if (i == f_cnt - 1)
+			if (i >= f_cnt - 2)
 				return CLI_CMD_SYNTAX;
 			match_key = find_column(f[++i]);
 			if (!match_key) {
-- 
1.7.4.1

From f9d7a5755e9df40409864667811a4497bbd75b63 Mon Sep 17 00:00:00 2001
From: Kozlov Dmitry <xeb@mail.ru>
Date: Fri, 2 Nov 2012 16:46:52 +0400
Subject: [PATCH] cli: fix double buffer free

---
 accel-pppd/cli/tcp.c    |    4 +++-
 accel-pppd/cli/telnet.c |    4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/accel-pppd/cli/tcp.c b/accel-pppd/cli/tcp.c
index 9ea914c..b2ed910 100644
--- a/accel-pppd/cli/tcp.c
+++ b/accel-pppd/cli/tcp.c
@@ -208,8 +208,10 @@ static int cln_write(struct triton_md_handler_t *h)
 		_free(cln->xmit_buf);
 		cln->xmit_pos = 0;
 
-		if (list_empty(&cln->xmit_queue))
+		if (list_empty(&cln->xmit_queue)) {
+			cln->xmit_buf = NULL;
 			break;
+		}
 
 		cln->xmit_buf = list_entry(cln->xmit_queue.next, typeof(*cln->xmit_buf), entry);
 		list_del(&cln->xmit_buf->entry);
diff --git a/accel-pppd/cli/telnet.c b/accel-pppd/cli/telnet.c
index 314ac9e..7409767 100644
--- a/accel-pppd/cli/telnet.c
+++ b/accel-pppd/cli/telnet.c
@@ -518,8 +518,10 @@ static int cln_write(struct triton_md_handler_t *h)
 		_free(cln->xmit_buf);
 		cln->xmit_pos = 0;
 
-		if (list_empty(&cln->xmit_queue))
+		if (list_empty(&cln->xmit_queue)) {
+			cln->xmit_buf = NULL;
 			break;
+		}
 
 		cln->xmit_buf = list_entry(cln->xmit_queue.next, typeof(*cln->xmit_buf), entry);
 		list_del(&cln->xmit_buf->entry);
-- 
1.7.4.1

